### PR TITLE
Update submitted column on workshop dashboard quick view

### PR DESCRIPTION
**Pull request by megcrenshaw**
The "Submitted" column was using the `created_at` field from the DB. But the `created_at` date for teacher applications will be the `save` date if that's the first thing the teacher does on their application. Now, the column uses the `date_applied` method, which overrides to find the first `unreviewed` status in the `status_log`.

Updated to say "Not yet submitted" on incomplete apps:
image

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/pd/application/teacher_application.rb [hunk 1]
```diff
@@ -128,6 +128,12 @@ def year
       application_year
     end
 
+    # Since teacher applications may be created on save before they are submitted, we cannot rely
+    # on the created_at field. When applications are submitted, they have status 'unreviewed'
+    def date_applied
+      status_log.find {|status_entry| status_entry["status"] == "unreviewed"}&.[]("at")&.to_date&.iso8601
```
Comment (megcrenshaw) [inline dashboard/app/models/pd/application/teacher_application.rb:134]: I hacked this together ... suggestions for making this easier to read and cleaner are welcome!

File: dashboard/test/models/pd/application/teacher_application_test.rb [hunk 2]
```diff
@@ -109,6 +109,17 @@ class TeacherApplicationTest < ActiveSupport::TestCase
...
+    test 'date_applied finds date of first unreviewed status' do
+      tomorrow = Date.tomorrow.to_time
+      application = create :pd_teacher_application, status: 'incomplete'
+      assert_nil application.date_applied
+
+      Timecop.freeze(tomorrow) do
+        application.update!(status: 'unreviewed')
+        assert_equal tomorrow, application.date_applied.to_time
+      end
+    end
```
Comment (davidsbailey) [inline dashboard/test/models/pd/application/teacher_application_test.rb:128]: nice test! I would suggest adding a test case for reopening / resubmitting.

Comment (megcrenshaw) [reply dashboard/test/models/pd/application/teacher_application_test.rb:128]: Good call! I modified this test for re-opening and re-submitting. Let me know if you think they should be separated out

File: dashboard/app/models/pd/application/teacher_application.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/models/pd/application/teacher_application.rb:134]: looks good! the unit test makes it clear what this is doing 👍

**Review by davidsbailey — APPROVED** [review]

**Pull request merged**